src/main/java/com/google/vinet/servlets/LoginServlet.java

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.vinet.servlets;
+
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.UserService;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that returns a login link, which when visited redirects the user to
+ * the provided redirection URL.
+ */
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  private UserService userService;
+
+  /**
+   * Construct a LoginServlet that depends on the default implementaion of UserService.
+   */
+  public LoginServlet() {
+    this.userService = UserServiceFactory.getUserService();
+  }
+
+  /**
+   * Construct a LoginServlet that depends on the provided implementation of UserService..
```

Comment:
Double full stop?

Reply:
thanks, fixed this now https://github.com/googleinterns/step226-2020/pull/6/commits/47bf78146a7a6a2d50d67de49932f64b7b9e5277